src/Routers/PublicAPIRouter.js

```diff
@@ -15,7 +15,7 @@ export class PublicAPIRouter extends PromiseRouter {
     const appId = req.params.appId;
     const config = Config.get(appId);
 
-    if (!config.publicServerURL) {
+    if (!config || !config.publicServerURL) {
```

Comment:
Would you be willing to update this to handle `!config` separately? We would like to throw an exception rather than return a standard 404. Something like this should do
```js
if (!config) {
    throw new Parse.Error(Parse.Error.OTHER_CAUSE, 'appId does not exist in config.');
}
```

Reply:
following in middleware.js if appId can't be found, this should be 
```
function invalidRequest(req, res) {
  res.status(403);
  res.end('{"error":"unauthorized"}');
}
```

Reply:
Ahh, didn't see that initially. @bryandel take flovilmart's suggestion for the proper error to report.